### DISCUSSION TITLE
Will fixed values of hyperparameters from first model be fixed in all the future refreshes?

### DISCUSSION THREAD

**Discussion by CJ2407** [facebookexperimental/Robyn#796] [Q&A]
Had to fix hyperparameters to certain values for most of the media channels to replicate a model result that our leadership chose in order to write the JSON file. Which wasn't saved originally. 

Now when I am running a Refresh model, I am noticing that parameters for those channels are again fixed i.e. model didn't iterate through any values for those channels. Does this mean, refresh model uses the same hyperparameters as initial build unless we re-build the model again?

If so, should I be concerned with the results that the Refresh model is generating?

![image](https://github.com/facebookexperimental/Robyn/assets/91983160/1eddba47-170b-4de4-b872-39f3124e87ed)

**Comment by gufengzhou** [comment]
When you fix a hyppar, we assume you do it out of a strong believe and thus treat it as truths, meaning not iterating. Conceptually speaking nothing to worry about

**Reply by CJ2407** [reply]
Thanks @gufengzhou , but the refresh model is resulting in 0 coefficients for 4 of paid media variables which our team is not happy about. Is this happening because I fixed my hyppar and now the refresh model is unable to iterate through and adjust to change in strategies in the refresh time period? Usually does the refresh model uses exact same lower and upper bounds for each hyppar as used in the initial model? Below is the statement I found in the documentation, but wanted to clarify - 
![image](https://github.com/facebookexperimental/Robyn/assets/91983160/d24f447d-dba9-4d37-b26e-7218959c893a)

**Reply by gufengzhou** [reply]
sorry for the delay, the discussion here is not as closely monitored as the issues. I understand that the instabil result could happen in refresh from time to time. we've just pushed a change to incur more stability in refresh (by putting more weight on the DECOMP objective function). you could try it out. also having smaller refresh steps will help too, because smaller steps means smaller hyperparameter range in refresh, which means more stability.
